core/src/Kokkos_MasterLock.hpp

```diff
@@ -67,7 +67,7 @@ namespace Experimental {
 // bool try_lock()
 //
 template <typename ExecutionSpace>
-class MasterLock;
+class KOKKOS_DEPRECATED MasterLock;
```

Comment:
```suggestion
class MasterLock;
```
should make `gcc` happy. In the end, only the specialization is usable anyway.
We can, of course, just also add a comment here.